src/library_sigs.js

```diff
@@ -0,0 +1,911 @@
+/* Auto-generated by tools/gen_sig_info.py. DO NOT EDIT. */
```

Comment:
Is there not a problem with having a single file, that it has to apply to all compilation modes and combinations of flags? Do signatures not change depending on flags? (I guess we have `p` for pointer size, but is there nothing else?)

Reply:
In general no they shouldn't change no, outside of the pointer-size thing.

For symbols that do change signature based on link flags (e.g. GLFW == 2 vs GLFW == 3) this solution does not yet work.  But we still have the fallback of manual `__sig` generation, and as far as I can tell GLFW is the only current example of functions having different sigs in different modes.

Reply:
If we set the sig manually, though, won't it conflict with the automatic generation (and hit the assertion added for that)? Or would we make the autogenerator ignore that sig?

Reply:
Yes, but we only do one or the other.  Either auto-generated, or manual, never both.   

This auto-generation covers about 800 of 1000 signatures, and I don't include ones that can differ with compilation mode such as GLFW which are still manual.

Reply:
I've got an error message that prints if `__sig` is specified in more than one place, we it will never be the case that we have both at the same time.

Reply:
See #19028 to see the effect removing just the ~800 symbol subset.